protocol-definitions/custom/Custom.yaml

```diff
@@ -780,6 +780,34 @@ customTypes:
         type: MemberVersion
         nullable: false
         since: 2.0
+      - name: addressMap
+        type: Map_EndpointQualifier_Address
+        nullable: false
+        since: 2.1
+  - name: EndpointQualifier
+    returnWithFactory: true
+    since: 2.1
+    params: 
+      - name: type
+        type: ProtocolType
+        nullable: false
+        since: 2.1
+      - name: identifier
+        type: String
+        since: 2.1
+        nullable: true
+  - name: ProtocolType
+    returnWithFactory: true
```

Comment:
do we need this?

Reply:
Yes, if we delete this line then the generated code will try to instantiate with the `new` operator, which won't compile, since `ProtocolType` is an enum. So we need to pass the two attributes and implement an enum lookup method in `CustomTypeFactory`. See [in this PR](https://github.com/hazelcast/hazelcast/pull/16910/files#diff-35eafc96056b27133382f0fb9da039a0R197)